### PR TITLE
Updates in Tony's code

### PR DESCRIPTION
**Pull request by siuwuncheung**
- Enable sine series coefficients computation
- Enable adaptive decay learning rate by `ReducedLRonPlateau`
- Add more choices of functions `g_0(x)` and `d(x)`

### CODE REVIEW

File: tony/test_poisson.py [hunk 1]
```diff
@@ -212,6 +218,15 @@ def run_experiment(config: PoissonSolverConfig, rank_val):
...
     elif config.lr_scheduler_type == 'MultiStepLR':
         lr_scheduler = optim.lr_scheduler.MultiStepLR(optimizer, milestones=config.lr_step_milestones, gamma=config.lr_step_gamma)
+    elif config.lr_scheduler_type == 'ReduceLROnPlateau':
+        lr_scheduler = ReduceLROnPlateau(
+            optimizer,
+            mode=config.lr_patience_mode,
+            factor=config.lr_factor,
+            patience=config.lr_patience,
+            min_lr=config.lr_min,
+            verbose=True
```
[the remark this answers is not in the document]
Comment (stevengogogo) [reply tony/test_poisson.py]: I have suggested  PR #21  to remove this error

**Review by stevengogogo — APPROVED** [review]
All modifications are under `tony/` folder, and the `test_poisson.py` is runnable with the arguments in `tony/run.tuo`.  This PR should be fine other PRs that change `pinn/` folder

**Review by liruipeng — APPROVED** [review]
